pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -577,46 +577,50 @@ func removeLibvirtCrcNetwork() error {
 }
 
 func removeCrcVM() error {
-	stdout, _, err := crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "domstate", constants.InstanceName())
-	if err != nil {
-		//  User may have run `crc delete` before `crc cleanup`
-		//  in that case there is no crc vm so return early.
-		return nil
-	}
-	if strings.TrimSpace(stdout) == "running" {
-		_, stderr, err := crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "destroy", constants.InstanceName())
+	for _, instanceName := range getAllInstanceNamesToDelete() {
```

Comment:
I'd prefer to see this commit split in 2 parts, one adjusting for the addition of the `preset` arg to the `constants.InstanceName()` function, and another one adding the iteration over multiple directories. 
Not sure we want the iteration over multiple dirs right now.

Reply:
This commit is also doing a lot more than what one might think by reading the commit log.

Reply:
> I'd prefer to see this commit split in 2 parts, one adjusting for the addition of the `preset` arg to the `constants.InstanceName()` function, and another one adding the iteration over multiple directories. 

that'd be tricky, as preflight checks don't have a way to know the current preset, we could pass the `config.Storage` to each preflight check.. but i added the iteration over machine dirs part to avoid this

> Not sure we want the iteration over multiple dirs right now.

if we don't allow more then one preset to run then we don't need to delete all the machine dirs right now, but this was a workaround for the above problem as well, this way we don't have to change the signature of `checkFunc` and since we'll need it later when having multiple presets running, i'd prefer it to add this now itself

Reply:
> that'd be tricky, as preflight checks don't have a way to know the current preset, we could pass the config.Storage to each preflight check.. 

There are already a few places where this is done. I'm not a huge fan of how it's done, but this works, see https://github.com/cfergeau/crc/commit/aec4e14c43f0314a33218feb4f909a4ac188d838